acorn/src/regexp.js

```diff
@@ -834,21 +866,50 @@ pp.regexp_eatLoneUnicodePropertyNameOrValue = function(state) {
 // https://www.ecma-international.org/ecma-262/8.0/#prod-CharacterClass
 pp.regexp_eatCharacterClass = function(state) {
   if (state.eat(0x5B /* [ */)) {
-    state.eat(0x5E /* ^ */)
-    this.regexp_classRanges(state)
+    const negate = state.eat(0x5E /* ^ */)
+    this.regexp_classContents(state)
     if (state.eat(0x5D /* ] */)) {
+      if (negate && state.lastMayContainStrings) {
+        state.raise("Negated character class may contain strings")
+      }
       return true
     }
+    if (state.switchV) {
```

Comment:
What is the idea behind this block? Does `regexp_nonEmptyClassRanges` as called through `regexp_classContents` not consume all the set content? What does the comment that says "Unreachable since ..." mean?

Reply:
This process is to issue the same "Invalid character in character class" message as V8.
Without this process, broken expressions such as `[A&&]` will display the message "Unterminated character class".

Reply:
Acorn as a whole does not try to mimic the precise error messages from browsers. I think it would be preferable to have less code and complexity (also [here](https://github.com/acornjs/acorn/pull/1218/files#diff-e9234a7f89243ee71abbff25886b1a2c8787112fbfb386264168adb288f0106dR1022)) rather than V8-aligned error messages.

Reply:
Thanks for your comment. I will remove those processes that have become too complicated.

Reply:
https://github.com/acornjs/acorn/pull/1219/commits/826f6f7af1ae2ea91e3105f89a7fe7df9bc9d3c4